rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -681,6 +685,16 @@ private String buildDeployingNodeURL(String pnName) {
 	}
 
 	/**
+	 * Called by the system every time a node that is DOWN reconnects
+	 * and changes its status to FREE or BUSY.
+	 *
+	 * @param node the node that has reconnected.
+	 */
+	public void onDownNodeReconnection(Node node) {
```

Comment:
I do not prefer. Some child classes do not need to implement it, this avoids some duplicated code.   Furthermore, in the near future, this method will probably contain code common to all infrastructures.